frontend/ccs.go

```diff
@@ -27,7 +29,8 @@ type CompiledConstraintSystem interface {
 	io.WriterTo
 	io.ReaderFrom
 
-	// IsSolved(witness witness.Witness, opts ...func(opt *backend.ProverOption) error) error
+	// IsSolved returns nil if given witness solves the constraint system and error otherwise
+	IsSolved(witness witness.Witness, opts ...func(opt *backend.ProverOption) error) error
```

Comment:
NewWitness returns pointer to witness.Witness - would it make sense to accept a pointer here instead?